### PR TITLE
chore(workflows): add `skywalking-eyes` license header check

### PR DESCRIPTION
**Pull request by homksei**
## Description

Added [`skywalking-eyes`](https://github.com/apache/skywalking-eyes) license header check:
-  `.github/workflows/skywalking-eyes.yml` workflow (actually, the check itself)
-  `.github/.licenserc.yaml` config for checker
--- 
### Additionally: 
Copyrights in some files were updated: 

- Added license header to the following `.md` files: 
  - `.github/ISSUE_TEMPLATE/RFC.md`
  - `.github/ISSUE_TEMPLATE/bug_report.md`
  - `.github/ISSUE_TEMPLATE/documentation.md`
  - `.github/ISSUE_TEMPLATE/feature_request.md`
  - `.github/pull_request_template.md`
  - `CODE_OF_CONDUCT.md`

- Removed second year in copyright notice for the following files:
  - `cpp/daal/src/externals/service_math_ref.h`
  - `cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl.hpp`
  - `cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_kernel_hist_impl_dpc.cpp`
  - `cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels.hpp`
  - `cpp/oneapi/dal/algo/decision_forest/backend/gpu/train_service_kernels_dpc.cpp`

- Fixed framing for the following `.rst` files (in `docs/source`): 
   - `docs/source/404.rst`
  - `docs/source/api/algorithms/clustering/dbscan.rst`
  - `docs/source/api/algorithms/clustering/index.rst`
  - `docs/source/api/algorithms/clustering/kmeans-init.rst`
  - `docs/source/api/algorithms/clustering/kmeans.rst`
  - `docs/source/api/algorithms/covariance/covariance.rst`
  - `docs/source/api/algorithms/covariance/index.rst`
  ...

### CODE REVIEW

File: .github/.licenserc.yaml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+header:
+  license:
+    spdx-id: Apache-2.0
+    copyright-owner: contributors to the oneDAL project
```
Comment (napetrov) [inline .github/.licenserc.yaml:18]: can it also check that older Intel copyright owner that still exist?

Comment (homksei) [reply .github/.licenserc.yaml]: removed this field so it won't be confusing, as it's not used for check.

**Review by napetrov — APPROVED** [review]
I would assume that overall check works and it would catch copyright that is not compliant

**Comment by homksei** [comment]
@Mergifyio rebase

File: .github/workflows/skywalking-eyes.yml [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+on:
+  pull_request:
+    branches:
+      - main
+  push:
+    branches:
+      - main
+
+permissions:
+  contents: read
```
Comment (icfaust) [inline .github/workflows/skywalking-eyes.yml:12]:
```suggestion
permissions: read-all
```

File: .github/workflows/skywalking-eyes.yml [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+  cancel-in-progress: true
+
+jobs:
+  check-license-header:
+    name: Copyright Check
+    if: github.repository == 'uxlfoundation/oneDAL'
+    runs-on: ubuntu-24.04
+    permissions:
+      issues: write
+      pull-requests: write
```
Comment (icfaust) [inline .github/workflows/skywalking-eyes.yml:25]: Same question as in uxlfoundation/scikit-learn-intelex#2215

**Review by icfaust — COMMENTED** [review]
This review mirrors this one: https://github.com/uxlfoundation/scikit-learn-intelex/pull/2215#pullrequestreview-2487759582

File: .github/workflows/skywalking-eyes.yml (hunk 3 above)
Comment (homksei) [reply .github/workflows/skywalking-eyes.yml:24]: The check can leave comments on the pull request if it finds issues. For this, write permissions need to be granted for these scopes.

**Review by icfaust — APPROVED** [review]

**Pull request merged**